tests/unit/test_util/test_security.py

```diff
@@ -0,0 +1,304 @@
+		return _helper
+
+	def setUp(self) -> None:
+		self._getWindowPatch = patch("winUser.getWindow", self._getWindow_patched)
+		self._getWindowPatch.start()
+		self._windows: List[winUser.HWNDVal] = list(range(1, 11))
+		"""
+		List of fake HWNDs, given an ordered index to make testing easier.
+		Must be 1 indexed as a HWND of 0 is treated an error.
+		"""
+		return super().setUp()
+
+	def tearDown(self) -> None:
+		self._getWindowPatch.stop()
+		return super().tearDown()
+
+
+class Test_isWindowAboveWindowMatchesCond_static(_Test_isWindowAboveWindowMatchesCond):
+	"""Test fetching a z-index when the order of window does not change"""
+	def test_windowNotFound(self):
+		startWindow = len(self._windows) // 2
+		with self.assertRaises(_WindowNotFoundError):
+			_isWindowAboveWindowMatchesCond(startWindow, lambda x: False)
+
+	def test_isAbove(self):
+		aboveIndex = 1
+		belowIndex = 2
+		self.assertTrue(_isWindowAboveWindowMatchesCond(aboveIndex, self._windowMatches(belowIndex)))
+
+	def test_isBelow(self):
+		aboveIndex = 1
+		belowIndex = 2
+		self.assertFalse(_isWindowAboveWindowMatchesCond(belowIndex, self._windowMatches(aboveIndex)))
+
+
+class Test_isWindowAboveWindowMatchesCond_dynamic(_Test_isWindowAboveWindowMatchesCond):
+	"""
+	Test fetching comparing the relative order of 2 windows,
+	where a window moves during the operation.
+
+	This test models changes when performing a bi-direction search that expects the
+	start window to be before/above the end window.
+	By symmetry, the same test results are expected if the search goes the other way,
+	expecting it to return False, rather than True.
+
+	To model changes in z-order, a _MoveWindow is used to describe the change.
+	When the getWindow is called with the triggerIndex, the value at start index is moved
+	"in front" of the window at end index.
+	Effectively this means that before getting the window at the triggerIndex, the order of
+	windows will change.
+	"""
+	_queuedMove: Optional[_MoveWindow] = None
+
+	def _getWindow_patched(self, hwnd: winUser.HWNDVal, relation: int) -> int:
+		self._moveIndexToNewIndexAtIndexOnce(self._windows.index(hwnd))
+		result = super()._getWindow_patched(hwnd, relation)
+		return result
+
+	def _moveIndexToNewIndexAtIndexOnce(self, currentIndex: int):
+		from logging import getLogger
+		
+		getLogger().error(f"Current index {currentIndex}, {self._windows}")
+		if (
+			self._queuedMove
+			and currentIndex == self._queuedMove.triggerIndex
+			and not self._queuedMove.triggered
+		):
+			self._queuedMove.triggered = True
+			window = self._windows.pop(self._queuedMove.startIndex)
+			self._windows.insert(self._queuedMove.endIndex, window)
+
+	def test_visited_windowMoves_pastTarget(self):
+		"""
+		A visited window is moved past the target window.
+		This does not affect the relative z-order.
+		"""
+		startWindow = 2
+		targetWindow = 5
+		self._queuedMove = _MoveWindow(
+			startIndex=3,
+			endIndex=6,
+			triggerIndex=4
+		)

```

Comment:
I think this would be easier to read if it were talking about `hwndVal`s rather than indexes. The `startWindow` / `targetWindow` using hwnds and this using the 1 based index value is confusing.

Reply:
I'm moving using HWNDs instead now